core/test/prediction/SurvivalPredictionStrategyTest.cpp

```diff
@@ -0,0 +1,156 @@
+#include "prediction/SurvivalPredictionStrategy.h"
+
+#include "catch.hpp"
+
+using namespace grf;
+
+TEST_CASE("Kaplan-Meier survival estimates are correct", "[survival], [prediction]") {
+  /*
+    The following test compares a survival curve estimate with ground truth from
+    the `survival` package. The data is generated with:
+
+    library(survival)
+    n <- 50
+    p <- 1
+    X <- matrix(rnorm(n * p), n, p)
+    latent.time <- -log(runif(n)) * exp(0.1 * X[, 1])
+    censor.time <- rexp(n)
+    Y <- pmin(latent.time, censor.time)
+    C <- as.integer(latent.time <= censor.time)
+    failure.times <- sort(unique(Y[C == 1]))
+    Y.relabeled <- findInterval(Y, failure.times, rightmost.closed = FALSE, all.inside = FALSE, left.open = FALSE)
+
+    kaplan.meier <- survfit(Surv(Y, C) ~ 1, data = data.frame(Y, C))
+
+    length(failure.times) # num_failures
+    dput(c(Y.relabeled, C)) # data_matrix
+    kaplan.meier.surv <- dput(summary(kaplan.meier)$surv) # expected_predictions
+  */
+
+  size_t num_failures = 24;
+  size_t num_rows = 50;
+  size_t num_cols = 2;
+  size_t outcome_index = 0;
+
+  std::vector<double> data_matrix = {
+    10L, 22L, 19L, 0L, 18L, 7L, 6L, 13L, 4L, 14L, 5L, 10L, 24L,
+    4L, 9L, 23L, 4L, 3L, 16L, 11L, 11L, 7L, 20L, 7L, 21L, 1L, 23L,
+    10L, 24L, 7L, 15L, 2L, 12L, 8L, 17L, 14L, 9L, 10L, 2L, 11L, 23L,
+    20L, 16L, 8L, 8L, 10L, 24L, 23L, 22L, 10L, 0L, 1L, 1L, 0L, 1L,
+    1L, 1L, 1L, 0L, 0L, 1L, 0L, 0L, 1L, 0L, 0L, 0L, 1L, 1L, 0L, 1L,
+    0L, 0L, 0L, 1L, 1L, 0L, 0L, 0L, 0L, 1L, 0L, 1L, 0L, 1L, 1L, 1L,
+    1L, 1L, 0L, 1L, 1L, 0L, 0L, 1L, 0L, 1L, 0L, 0L, 0L
+  };
+
+  std::vector<double> expected_predictions = {
+    0.979591836734694, 0.959183673469388, 0.938331854480923, 0.917480035492458,
+    0.895635272742637, 0.873790509992817, 0.851945747242997, 0.828280587597358,
+    0.803181175851983, 0.77727210566321, 0.746181221436681, 0.712263893189559,
+    0.678346564942438, 0.644429236695316, 0.608627612434465, 0.572825988173614,
+    0.53463758896204, 0.496449189750465, 0.458260790538891, 0.420072391327317,
+    0.378065152194585, 0.336057913061853, 0.288049639767303, 0.192033093178202
+  };
+
+  DefaultData data(data_matrix, num_rows, num_cols);
+  data.set_outcome_index(outcome_index);
+  data.set_censor_index(outcome_index + 1);
+
+  std::vector<std::vector<size_t>> leaf_samples(1);
+  for (size_t i = 0; i < num_rows; i++) {
+    leaf_samples[0].push_back(i);
+  }
+
+  SurvivalPredictionStrategy prediction_strategy(num_failures);
+  PredictionValues prediction_values = prediction_strategy.precompute_prediction_values(leaf_samples, data);
+
+  std::vector<double> predictions = prediction_values.get_all_values()[0];
+  for (size_t i = 0; i < predictions.size(); i++) {
+    REQUIRE(equal_doubles(predictions[i], expected_predictions[i], 1e-10));
+  }
+}
+
+TEST_CASE("Kaplan-Meier estimates on duplicated data is the same as with sample weights equal to two", "[survival], [prediction]") {
```

Comment:
I'm a bit confused as to why this test refers to sample weights when they aren't implemented yet for survival forests ?

Reply:
No sample weights are implemented for prediction, but not splitting.

Reply:
Thanks for the explanation, I read too fast.